### PR TITLE
add DefaultClusterTemplate to seed

### LINKED ISSUE #7958: Support cni defaults for seeds
Support setting cni configuration for a seed

### PR DESCRIPTION
**Pull request by mate4st**
**What this PR does / why we need it**:

Provides a way to define default for new clusters on a template base.

**Which issue(s) this PR fixes**
Fixes #7958

```release-note
Provide functionality to define a default cluster template for new clusters per seed.
DefaultComponentSettings of the Seed object is deprecated in favor of DefaultClusterTemplate.
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/cluster_templates.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ const (
 	UserClusterTemplateScope    = "user"
 	ProjectClusterTemplateScope = "project"
 	GlobalClusterTemplateScope  = "global"
+	SeedTemplateScope           = "seed"
```
Comment (kron4eg) [inline pkg/apis/kubermatic/v1/cluster_templates.go]: This is new shadow API, it's not being used at the moment anywhere. Are you sure you wanted this change in here?

Comment (mate4st) [reply pkg/apis/kubermatic/v1/cluster_templates.go]: Yeah it has no effect at the mment, but let me check how the api uses these scopes.

File: cmd/seed-controller-manager/main.go [hunk 2]
```diff
@@ -28,10 +28,13 @@ import (
 	gatekeeperv1beta1 "github.com/open-policy-agent/frameworks/constraint/pkg/apis/templates/v1beta1"
 	"github.com/prometheus/client_golang/prometheus"
 	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
```
Comment (kron4eg) [inline cmd/seed-controller-manager/main.go]: please run https://github.com/xrstf/gimps

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**